tests/test_unit.py

```diff
@@ -6623,6 +6711,20 @@ def testWarnUsingLshiftForward(self):
             except Exception as e:
                 self.fail("raised warning when it should not have")
 
+    def testAssertParseAndCheckDict(self):
```

Comment:
Interesting, testing the test framework. I guess this particular assert hasn't been used anywhere else. Probably would be good some time to add unit tests for the others in pyparsing_test (though would not increase coverage percentage). Just a note to self, not a necessary thing to do right now.